socorro/external/es/crashstorage.py

```diff
@@ -138,22 +143,91 @@ def _submit_crash_to_elasticsearch(self, connection, crash_document):
+                    # Elasticsearch or just not a number.
+                    matches = re.findall(
+                        FIELD_NAME_NUMBER_ERROR_REGEX, e.error
+                    )
+                    if matches:
+                        field_name = matches[0]
+                else:
+                    # We don't know how to fix this exception. Let it raise.
+                    self.config.logger.critical(
+                        'Submission to Elasticsearch failed for %s (%s)',
+                        crash_id,
+                        e,
+                        exc_info=True
+                    )
+                    raise
 
-        except elasticsearch.exceptions.ElasticsearchException as e:
-            self.config.logger.critical(
-                'Submission to Elasticsearch failed for %s (%s)',
-                crash_id,
-                e,
-                exc_info=True
-            )
-            raise
+                if not field_name:
+                    # We are unable to parse which field to remove, we cannot
+                    # try to fix the document. Let it raise.
+                    self.config.logger.critical(
+                        'Submission to Elasticsearch failed for %s (%s)',
+                        crash_id,
+                        e,
+                        exc_info=True
+                    )
+                    raise
+
+                if field_name.endswith('.full'):
+                    # Remove the `.full` at the end, that is a special mapping
+                    # construct that is not part of the real field name.
+                    field_name = field_name[:-5]
+
+                # Now remove that field from the document before trying again.
+                field_path = field_name.split('.')
+                parent = crash_document
+                for i, field in enumerate(field_path):
+                    if i == len(field_path) - 1:
+                        # This is the last level, so `field` contains the name
+                        # of the field that we want to remove from `parent`.
+                        del parent[field]
+                    else:
+                        parent = parent[field]
+
+                # Add a note in the document that a field has been removed.
+                if crash_document.get('removed_fields'):
+                    crash_document['removed_fields'] = '{} {}'.format(
+                        crash_document['removed_fields'],
+                        field_name
+                    )
+                else:
+                    crash_document['removed_fields'] = field_name
+            except elasticsearch.exceptions.ElasticsearchException as e:
+                self.config.logger.critical(
+                    'Submission to Elasticsearch failed for %s (%s)',
+                    crash_id,
+                    e,
+                    exc_info=True
+                )
```

Comment:
Why is the point of this? I'm not trying to convince you otherwise, just curious. And if there's a good reason to log, then it would be nice to have a comment why this logging has to happen. 
We have Sentry now so the exception will be there and it will have the crash_id in it too.